GpsDataAnalyzer/fileparser/fileparser.py

```diff
@@ -0,0 +1,105 @@
+"""
+Usage: xmlparser.py
+
+Parse the xml or csv file and generate a GpsDataSet
+"""
+import xml.etree.ElementTree as ET
+
+from GpsDataAnalyzer.gpsdataset.gpsdataset import GpsData
+from GpsDataAnalyzer.gpsdataset.gpsdataset import GpsMetaData
+from GpsDataAnalyzer.gpsdataset.gpsdataset import GpsDataSet
+
+class FileParser:   
+    def __init__(self, filename):
+        """
+        Constructor of fileparser
+
+        Args:
+          filename: the filename //Could be replaced by file path
+        """
+        self.filename = filename
+
+    def get_file_type(self) -> str:
+        """
+        Get the file type
+        """
+        # if filename replaced by file path, should use os.path.splittext(filepath)
+        file_type = self.filename.split(".")[-1]
+
+        if file_type is None:
+            print("File extension is None!")
+            return None
+
+        return file_type
+
+
+    def parse_file(self) -> GpsDataSet:
+        """
```

Comment:
I think that since you only need one FileParser() to parse multiple different types of files, it makes sense to just initialize a FileParser() without a filename attribute, and then pass the file name into the parse_file method instead, which would return the parsed data for that file and could be called with any file.